cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -1431,17 +1762,116 @@ instance (Show addr, Show versionNumber, Show agreedOptions, ToObject addr,
+             , "policy" .= toObject _verb policyTrace
+             ]
+  toObject _verb (TrServerStarted peerAddrs)       =
+    mkObject [ "kind" .= String "AcceptPolicyTrace"
+             , "addresses" .= toJSON (toObject _verb `map` peerAddrs)
+             ]
+  toObject _verb TrServerStopped                   =
+    mkObject [ "kind" .= String "ServerStopped"
+             ]
+  toObject _verb (TrServerError exception)         =
+    mkObject [ "kind" .= String "ServerError"
+             , "error" .= show exception
+             ]
+
+instance ToJSON MiniProtocolNum where
+  toJSON (MiniProtocolNum w) =
+    Aeson.object [ "kind" .= String "MiniProtocolNum"
+                 , "num" .= w
+                 ]
 
 instance Show addr
       => ToObject (InboundGovernorTrace addr) where
-  -- TODO: a better 'ToObject' instance
-  toObject _verb ev =
-    mkObject [ "kind" .= String "InboundGovernorTrace"
-             , "event" .= show ev ]
+  toObject _verb (TrNewConnection p caddr)            =
+    mkObject [ "kind" .= String "NewConnection"
+             , "provenance" .= show p
+             , "connectionId" .= toObject _verb caddr
+             ]
+  toObject _verb (TrResponderRestarted caddr m)       =
+    mkObject [ "kind" .= String "ResponderStarted"
+             , "connectionId" .= toObject _verb caddr
+             , "miniProtocolNum" .= toJSON m
+             ]
+  toObject _verb (TrResponderStartFailure caddr m s)  =
+    mkObject [ "kind" .= String "ResponderStartFailure"
+             , "connectionId" .= toObject _verb caddr
+             , "miniProtocolNum" .= toJSON m
+             , "error" .= show s
+             ]
+  toObject _verb (TrResponderErrored caddr m s)       =
+    mkObject [ "kind" .= String "ResponderErrored"
+             , "connectionId" .= toObject _verb caddr
+             , "miniProtocolNum" .= toJSON m
+             , "error" .= show s
+             ]
+  toObject _verb (TrResponderStarted caddr m)         =
+    mkObject [ "kind" .= String "ResponderStarted"
+             , "connectionId" .= toObject _verb caddr
+             , "miniProtocolNum" .= toJSON m
+             ]
+  toObject _verb (TrResponderTerminated caddr m)      =
+    mkObject [ "kind" .= String "ResponderTerminated"
+             , "connectionId" .= toObject _verb caddr
+             , "miniProtocolNum" .= toJSON m
+             ]
+  toObject _verb (TrPromotedToWarmRemote caddr (UnsupportedState as)) =
+    mkObject [ "kind" .= String "PromotedToWarmRemote"
+             , "connectionId" .= toObject _verb caddr
+             , "result" .=
+                 Aeson.object [ "unsupportedState" .= toJSON as ]
+             ]
+  toObject _verb (TrPromotedToWarmRemote caddr (OperationSuccess as)) =
+    mkObject [ "kind" .= String "PromotedToWarmRemote"
+             , "connectionId" .= toObject _verb caddr
+             , "result" .=
+                 Aeson.object [ "operationSuccess" .= toJSON as ]
+             ]
```

Comment:
If we provide `ToJSON` instance for `OperationResult` we will be able to merge these two pattern matches into one (and a few similar cases below).